cfme/utils/events.py

```diff
@@ -257,73 +145,76 @@ def matches(self, evt):
             return True
 
     def add_attrs(self, *attrs):
+        """ Adds an EventAttr to event.
+
+        This method allows to add an attribute that are in :py:const:``DEFAULT_ATTRS`` to event.
         """
-        event consists of attributes like event_type, etc.
-        this method allows to add an attribute to event
-        """
-        if isinstance(attrs, Iterable):
-            for attr in attrs:
-                if attr.name == 'target_name':
-                    # this is artificial attr which will be converted to target_id during matching
-                    self.event_attrs[attr.name] = attr
-                elif attr.name in self._default_attrs:
-                    # type check was removed because sqlalchemy's python_type
-                    # and type of returned values are different
-                    self.event_attrs[attr.name] = attr
-                else:
-                    logger.warning('The attribute {} type {} is absent in DB '
-                                   'or type mismatch.'.format(attr.name, attr.type))
-        else:
+        if not isinstance(attrs, Iterable):
             raise ValueError("incorrect parameters are passed {}".format(attrs))
+
+        for attr in attrs:
+            if attr.name == 'target_name':
+                # This attribute will be converted to target_id during matching.
+                self.event_attrs[attr.name] = attr
+            elif attr.name in self.DEFAULT_ATTRS:
+                self.event_attrs[attr.name] = attr
+            else:
+                logger.warning('The attribute {} is absent in default attributes. '
+                               'Ignoring.'.format(attr.name))
         return self
 
-    def build_from_raw_event(self, evt):
-        """
-        helper method which takes raw event from event_streams and prepares event object
-        """
-        # checking is this param - raw event, populating fields by this data then
-        if self._is_raw_event(evt):
-            self._parse_raw_event(evt)
+    def build_from_entity(self, event_entity):
+        for key in self.DEFAULT_ATTRS:
+            self.add_attrs(EventAttr(**{key: getattr(event_entity, key, None)}))
         return self
 
 
 class EventListener(Thread):
+    """ EventListener accepts "expected" events, listens to db events and compares matched events
+    with expected events. Runs callback function if expected events have it.
+
+    :var FILTER_ATTRS: List of filters used in REST API call
     """
-     accepts "expected" events, listens to db events and compares showed up events with expected
-     events. Runs callback function if expected events have it.
-    """
+    FILTER_ATTRS = [
+        'event_type',
+        'target_type',
+        'target_id',
+        'source'
+    ]
+
     def __init__(self, appliance):
         super(EventListener, self).__init__()
         self._appliance = appliance
         self._tool = EventTool(self._appliance)
```

Comment:
are you sure you need event tool at all ?
I had to use it because of db connection but you are not tied to it.